crates/evm/src/hooks.rs

```diff
@@ -157,11 +177,26 @@ where
     ) {
         let l1_hash = soft_confirmation_info.da_slot_hash;
 
-        let parent_block = self
-            .head
-            .get(working_set)
-            .expect("Head block should always be set")
-            .seal();
+        let current_spec = soft_confirmation_info.current_spec;
+
+        let parent_block = if current_spec >= CitreaSpecId::Fork1 {
+            match self.head_rlp.get(working_set) {
+                Some(block) => block.seal(),
+                None => {
+                    let block: Block<AlloyHeader> = self
+                        .head
+                        .get(working_set)
+                        .expect("Head block should always be set")
+                        .into();
+                    block.seal()
+                }
+            }
+        } else {
+            self.head
+                .get(working_set)
+                .expect("Head block should always be set")
+                .seal()
+        };
```

Comment:
want to make a function for this? same stuff happens in two places

Reply:
I tried it but it made worse, created more copying because of having to do more from into conversions, can stay as is